packages/autorest.go/src/generator/fake/internal.ts

```diff
@@ -85,3 +106,47 @@ func contains[T comparable](s []T, v T) bool {
 	return false
 }
 `;
+
+const tracker = `
+func newTracker[T any]() *tracker[T] {
+	return &tracker[T]{
+		items: map[string]*T{},
+	}
+}
+
+type tracker[T any] struct {
+	items map[string]*T
+	mu sync.Mutex
+}
+
+func (p *tracker[T]) key(req *http.Request) string {
+	path := req.URL.Path
```

Comment:
This assumes that path will be unique per invocation.  Probably always true for ARM, but is it universally true?  I'm thinking perhaps a `POST` for the same path but with differing query parameters.

Reply:
Unfortunately we don't propagate query parameters when creating the fake poller or injecting next links for paged operations.  Will fix this in the next `azcore` beta.